src/reducers/tokens/index.js

```diff
@@ -0,0 +1,73 @@
+import { handleActions } from 'redux-actions'
+import reduceReducers from 'reduce-reducers'
+
+import { tokens } from '../../actions/tokens'
+
+const initialState = {
+    likelyContracts: [],
+    tokens: {}
+}
+
+const likelyContractsReducer = handleActions({
+    [tokens.likelyContracts.get]: (state, { ready, error, payload }) =>
+        (!ready || error)
+            ? state
+            : ({
+                ...state,
+                likelyContracts: payload
+            }),
+}, initialState)
+
+const tokensReducer = handleActions({
+    [tokens.likelyContracts.get]: (state, { ready, error, payload, meta }) => 
+        (!ready || error)
+            ? state
+            : ({
+                ...state,
+                tokens: [...payload, ...meta].reduce((x, contract) => ({
+                    ...x,
+                    [contract]: { 
+                        contract 
```

Comment:
>We can assume that contract is the string representation of the contract name; can we call it contractName so it's apparent that it's not a rich object without looking at how we use it and/or where it comes from?

Agree, renamed everywhere.

Reply:
>Also, curious about why we are creating an object keyed with the name of contract, but with only one key named contract that is the same as the containing object's key; what is the purpose of the nested contract string?

Right, refactored it a bit - now we will save likelyTokens as a separate object, but at the same time we will keep details for tokens with metadata only. It will be much simpler.